clusterloader2/pkg/test/test.go

```diff
@@ -37,41 +38,77 @@ var (
 	Test = createSimpleExecutor()
 )
 
-// RunTest runs test based on provided test configuration.
-func RunTest(
+// CreateTestContext creates the test context.
+func CreateTestContext(
 	clusterFramework *framework.Framework,
 	prometheusFramework *framework.Framework,
 	clusterLoaderConfig *config.ClusterLoaderConfig,
 	testReporter Reporter,
-) *errors.ErrorList {
+	testScenario *api.TestScenario,
+) (Context, *errors.ErrorList) {
 	if clusterFramework == nil {
-		return errors.NewErrorList(fmt.Errorf("framework must be provided"))
+		return nil, errors.NewErrorList(fmt.Errorf("framework must be provided"))
 	}
 	if clusterLoaderConfig == nil {
-		return errors.NewErrorList(fmt.Errorf("cluster loader config must be provided"))
+		return nil, errors.NewErrorList(fmt.Errorf("cluster loader config must be provided"))
 	}
 	if CreateContext == nil {
-		return errors.NewErrorList(fmt.Errorf("no CreateContext function installed"))
+		return nil, errors.NewErrorList(fmt.Errorf("no CreateContext function installed"))
+	}
+
+	mapping, errList := config.GetMapping(clusterLoaderConfig, testScenario.OverridePaths)
+	if errList != nil {
+		return nil, errList
 	}
+
+	return CreateContext(clusterLoaderConfig, clusterFramework, prometheusFramework, state.NewState(), testReporter, mapping, testScenario), errors.NewErrorList()
+}
+
+// CompileTestConfig loads the test configuration and nested modules.
+func CompileTestConfig(ctx Context) (*api.Config, *errors.ErrorList) {
 	if Test == nil {
-		return errors.NewErrorList(fmt.Errorf("no Test installed"))
+		return &api.Config{}, errors.NewErrorList(fmt.Errorf("no Test installed"))
 	}
 
-	mapping, errList := config.GetMapping(clusterLoaderConfig)
-	if errList != nil {
-		return errList
+	clusterLoaderConfig := ctx.GetClusterLoaderConfig()
+	testConfigFilename := filepath.Base(ctx.GetTestScenario().ConfigPath)
+	testConfig, err := ctx.GetTemplateProvider().TemplateToConfig(testConfigFilename, ctx.GetTemplateMappingCopy())
+	if err != nil {
+		return &api.Config{}, errors.NewErrorList(fmt.Errorf("config reading error: %v", err))
```

Comment:
return nil *api.Config on errors, here and in the other places

Reply:
Addressed